crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -1,23 +1,53 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use async_trait::async_trait;
+use diesel::r2d2::R2D2Connection;
+use futures::StreamExt;
+use mysten_metrics::get_metrics;
+use mysten_metrics::metered_channel::{Receiver, Sender};
+use mysten_metrics::spawn_monitored_task;
+use prometheus::Registry;
+use sui_data_ingestion_core::DataIngestionMetrics;
+use sui_data_ingestion_core::IndexerExecutor;
+use sui_data_ingestion_core::ReaderOptions;
+use sui_data_ingestion_core::ShimProgressStore;
+use sui_data_ingestion_core::Worker;
+use sui_data_ingestion_core::WorkerPool;
+use sui_package_resolver::Resolver;
+use sui_rest_api::CheckpointData;
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+use tokio::sync::oneshot;
+use tokio::sync::watch;
 use tokio_util::sync::CancellationToken;
 use tracing::info;
 
-use sui_rest_api::Client;
-
+use crate::store::package_resolver::IndexerStorePackageResolver;
+use crate::store::package_resolver::InterimPackageResolver;
 use crate::types::IndexerResult;
+use crate::IndexerConfig;
 use crate::{metrics::IndexerMetrics, store::IndexerStore};
+use std::sync::{Arc, Mutex};
+use sui_package_resolver::PackageStoreWithLruCache;
+
+use super::checkpoint_handler::CheckpointHandler;
+use super::tx_processor::IndexingPackageBuffer;
+use super::TransactionObjectChangesToCommit;
 
 const OBJECTS_SNAPSHOT_MAX_CHECKPOINT_LAG: usize = 900;
```

Comment:
thinking twice about the max_lag, I originally pick 900 and 300 to avoid frequent query init overhead, and it's great that we do not have it any more, I think we can get rid of max_lag and limit the commit batch size by a separate const, so that 1. each commit task is smaller and less spiky 2. expected lag is less 3. less mem usage. Meanwhile the "guaranteed" available range is still 300, wdyt?

Reply:
Sounds good. We just need one threshold for determining when to update objects_snapshot. And this lag can be something like 300 or 300 + batch_size, if we want the available range to be at least 300.